src/core/utils/Utilities.ts

```diff
@@ -170,3 +172,32 @@ export const decodeBlock = (input: any, inputOffset: number, output: any, output
     output[outputOffset + 3] = ((bytes[4] & 0x01) << 7) | (bytes[5] << 2) | (bytes[6] >> 3);
     output[outputOffset + 4] = ((bytes[6] & 0x07) << 5) | bytes[7];
 };
+
+/**
+ * Generate keccak 512 hash given by data.
+ * @param {Uint8Array} data The data to hash.
+ * @returns {Uint8Array} The keccak hash value.
+ * */
+export const keccakHash = (data: Uint8Array): Uint8Array => {
+    const hasher = new Keccak(512);
+    const hex = hasher.update(Buffer.from(data)).digest('hex');
+    return Converter.hexToUint8(hex);
+};
+
+/**
+ * Create keccak 512 Hasher object used to hash data.
+ * @returns KeccakHasher object
+ **/
+export const KeccakHasher = () => {
+    const hasher = new Keccak(512);
+    return {
+        update: (data) => {
+            hasher.update(Buffer.from(data));
+        },
+        digest: () => {
+            let hash = hasher.digest('hex');
+            return Converter.hexToUint8(hash);
```

Comment:
same here

Reply:
don't think we need this hasher lambda at all? If using `sha3` you can call: `new Keccak(512).update(..).digest()` instead

Reply:
nope you don't need it (hasher), but you _probably_ want to wrap the result in uin8array

Reply:
This object is purely used inside `nacl-fast.js`'s custom method.
In `nacl-fast`'s `reduce()` `scalarbase()` use `uint8Array` as params.

`digest()` is return into Buffer rather than `uint8Array`.

will dig more